src/Optimizely/Config/DatafileProjectConfig.php

```diff
@@ -230,19 +273,19 @@ public function __construct($datafile, $logger, $errorHandler)
 
         $this->_accountId = $config['accountId'];
         $this->_projectId = $config['projectId'];
-        $this->_anonymizeIP = isset($config['anonymizeIP'])? $config['anonymizeIP'] : false;
-        $this->_botFiltering = isset($config['botFiltering'])? $config['botFiltering'] : null;
+        $this->attributes = $config['attributes'] ?: [];
+        $this->audiences = $config['audiences'] ?: [];
+        $this->events = $config['events'] ?: [];
+        $this->typedAudiences = isset($config['typedAudiences']) ? $config['typedAudiences'] : [];
```

Comment:
why this can't be $config['typedAudiences'] ?: []

Reply:
i just followed the existing way to get typedAudiences ealier it was being used as a variable but i made it a class attribute.